### PR TITLE
Fix incorrect downloaded file size when size of the file is not a multiple of the block size

### LINKED ISSUE #21995: The downloaded file won't be truncated to correct size after sdk/storage/azblob:v1.2.0
### Bug Report

- import path of package in question, e.g. `github.com/Azure/azure-sdk-for-go/sdk/storage/azblob`
- SDK version e.g. `v1.2.0`
  - Specify the exact commit if possible: https://github.com/Azure/azure-sdk-for-go/pull/21259
- output of `go version`: 1.20

- What happened?
The downloaded file will not be truncated after method `DownloadFile` under `azure-sdk-for-go/sdk/storage/azblob/blockblob/client.go` is called. Therefore the md5 and size of the downloaded file does not match the original blob in container.
- What did you expect or want to happen?
  The file size and md5 should be the same between the downloaded file and the original one in the container.
- How can we reproduce it?
   Call the method `DownloadFile` and compare the sha/md5 and the size of the file.

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
Fixes #21995 

Whenever the user downloads a file and the size of the file is not a multiple of the block size he has given, the size of downloaded file is greater than the actual file. 
contents are - actual contents + empty spaces in the end

https://github.com/Azure/azure-sdk-for-go/blob/c0bce1155328b0aa91d2bcb996f971470b89923e/sdk/storage/azblob/blob/client.go#L535C1-L543C14

If block size is 2 MB, and file size was 7 MB, the last 1 MB of data left is lesser than our block size. 
We are reading  the right size of data to the buffer, but while writing, we are writing the entire buffer and not the data we read - which leads to writing extra spaces to the end of the file.

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]

**Review by nakulkar-msft — APPROVED** [review]

File: sdk/storage/azblob/blob/client.go [hunk 1]
```diff
@@ -489,20 +489,22 @@ func (b *Client) downloadFile(ctx context.Context, writer io.Writer, o downloadO
...
 	writerError := make(chan error)
+	writeSize := int64(0)
 	go func(ch chan error) {
 		for _, block := range blocks {
 			select {
 			case <-ctx.Done():
 				return
 			case block := <-block:
 				_, err := writer.Write(block)
+				writeSize += int64(len(block))
```
[the remark this answers is not in the document]
Comment (nakulkar-msft) [reply sdk/storage/azblob/blob/client.go:507]: We can also do
```
n, err := writer.Write(block)
writeSize += int64(n)
```

[the remark this answers is not in the document]
Comment (souravgupta-msft) [reply sdk/storage/azblob/blob/client.go]: Done

**Pull request merged**